modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaManager.java

```diff
@@ -1005,6 +1021,49 @@ public boolean isReplicaStarted(ReplicationGroupId replicaGrpId) {
         return replicaFuture != null && isCompletedSuccessfully(replicaFuture);
     }
 
+    /**
+     * Can possibly start replica if it's not running or is stopping.
+     *
+     * @param groupId Group id.
+     * @param startOperation Replica start operation.
+     * @param forcedAssignments Assignments to reset forcibly, if needed. Assignments reset is only available when replica is started.
+     */
+    public CompletableFuture<Boolean> weakReplicaStart(
```

Comment:
Didn't check that part with my own eyes. Do we update client peers and learners outside the closure? I mean that we should.